### ISSUE TITLE
Using SQL and MongoDB in the same project

### ISSUE THREAD

**Issue by luisp88** [abpframework/abp#3489]
I downloaded a project template for EFc and I'm using SQL Server OK.
I want to use both, SQLS and MongoDB, so created another project with its ..DbModule.cs and ..DbContext.cs
Added
![image](https://user-images.githubusercontent.com/4925069/78613036-c8c3d380-7838-11ea-9fe5-d50ccbda206a.png)
in DependsOn to ProjectDbMigratorModule, so breakpoint hitting AddMongoDbContext<ProjectMongoDbContext> OK.
![image](https://user-images.githubusercontent.com/4925069/78613080-df6a2a80-7838-11ea-8527-2643b2f47ba2.png)
Declared one FullAuditedAggregateRoot<> and added as IMongoCollection
![image](https://user-images.githubusercontent.com/4925069/78612946-8c907300-7838-11ea-8174-972d31dd898e.png)
Also have the connection string in .DbMigrator's appsettings.json

Problem: No database creation! Downloaded MongoDB project template, changed the connection string and that project does create the database :(

**Comment by hitaspdotnet** [comment]
Required services was replaced by each other.

**Comment by luisp88** [comment]
I there anyway  to use abp to add both services without using services.Replace() in abp's AddXDbContext?

**Comment by maliming** [comment]
@luisp88  

I think EF Core and Mongodb cannot work together in the same project.

It can be used when deploying the module as a remote service.

**Comment by luisp88** [comment]
If someone wants to use dbentities and mongocollections in same solution and inject "sql entities" and "mongo collections" in same appservice:
What I just did was to create another project similar as EntityFrameworkCore but for MongoDB. Added it to DependsOn in DbMigratorModule and HttpApiHostModule (keeped DbMigrations clean without dependson MongoDb).
In the new project module, added in ConfigureServices:
`context.Services.AddMongoDbContext<FichaMedicaMongoDbContext>(options =>
            {
                options.AddDefaultRepositories(includeAllEntities: true);
            });`
and also DependsOn AbpMongoDbModule.
Declared a collection in its MongoDbContext, added dataseeders and everything works ok.

One question: each time I run the app and use some collection (.CoutnAsync .ToList or InsertAsync), the first executions always enters on .CreateModel of my new MongoDbContext class (with dbcontext that doesn't happen). Is it ok?
